<launch>

  <arg name="uav_name" default="$(optenv UAV_NAME uav)"/>
  <arg name="profiler" default="$(optenv PROFILER false)" />

  <!-- launch the node -->
  <node name="serial_gripper" pkg="mrs_serial" type="baca_protocol" output="screen" respawn="true">

    <rosparam file="$(find mrs_serial)/config/mrs_serial.yaml" />

    <param name="enable_profiler" type="bool" value="$(arg profiler)" />
    <param name="portname" value="/dev/gripper"/>
    <param name="use_timeout" value="true"/>
    <param name="swap_garmins" value="$(optenv SWAP_GARMINS false)"/>

    <!-- Publishers -->
    <remap from="~range" to="/$(arg uav_name)/garmin/range" />
    <remap from="~range_up" to="/$(arg uav_name)/garmin/range_up" />
    <remap from="~profiler" to="profiler" />
    <remap from="~baca_protocol_out" to="~received_message" />

    <!--ServicesIn-->
    <remap from="~netgun_fire" to="grip" />
    <remap from="~netgun_arm" to="netgun_arm" />
    <remap from="~netgun_safe" to="ungrip" />
    <remap from="~gimbal_pitch" to="gimbal/set_pitch" />

    <!-- Subscribers -->
    <remap from="~magnet_in" to="ungrip" />
    <remap from="~baca_protocol_in" to="~send_message" />

  </node>

</group>

</launch>
